core/domain/collection_jobs_one_off_test.py

```diff
@@ -133,143 +137,466 @@ def test_migrate_collections_failing_strict_validation(self):
+            migrated_collection_rights_model.last_updated)
 
-        output = (
-            collection_jobs_one_off.CollectionMigrationOneOffJob.get_output(
-                job_id))
-        expected = [[u'validation_error',
-                     [u'Collection %s failed validation: Invalid '
-                      u'language code: %s'
-                      % (self.COLLECTION_ID, collection_language_code)]]]
-        self.assertEqual(expected, [ast.literal_eval(x) for x in output])
 
-    def test_migration_job_migrates_collection_nodes(self):
-        """Tests that the collection migration job migrates content from
-        nodes to collection_contents if collection_contents is empty.
+class MockCollectionModel(collection_models.CollectionModel):
+    """Mock CollectionModel so that it allows to set `nodes`."""
+
+    nodes = datastore_services.JsonProperty(default={}, indexed=False)
+
+    def _trusted_commit(
+            self, committer_id, commit_type, commit_message, commit_cmds):
+        """Record the event to the commit log after the model commit.
+
+        Note that this extends the superclass method.
+
+        Args:
+            committer_id: str. The user_id of the user who committed the
+                change.
+            commit_type: str. The type of commit. Possible values are in
+                core.storage.base_models.COMMIT_TYPE_CHOICES.
+            commit_message: str. The commit description message.
+            commit_cmds: list(dict). A list of commands, describing changes
+                made in this model, which should give sufficient information to
+                reconstruct the commit. Each dict always contains:
+                    cmd: str. Unique command.
+                and then additional arguments for that command.
         """
-        # Create an exploration to put in the collection.
-        self.save_new_default_exploration(self.EXP_ID, self.albert_id)
-        node = collection_domain.CollectionNode.create_default_node(self.EXP_ID)
-
-        # Create a collection directly using the model, so that the collection
-        # nodes are stored in the 'nodes' property rather than the
-        # 'collection_contents' property.
-        collection_title = 'A title'
-        collection_category = 'A category'
+        base_models.VersionedModel._trusted_commit(  # pylint: disable=protected-access
+            self, committer_id, commit_type, commit_message, commit_cmds)
+
+
+class RemoveCollectionModelNodesOneOffJobTests(test_utils.GenericTestBase):
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    COLLECTION_ID = 'collection_id'
+
+    def setUp(self):
+        super(RemoveCollectionModelNodesOneOffJobTests, self).setUp()
+
+        # Setup user who will own the test objects.
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.process_and_flush_pending_mapreduce_tasks()
```

Comment:
@iramin I would try removing this, I have a slight suspicion that it leads to the errors.

Reply:
@vojtechjelinek 
It is used in other tests that don't fail. for example:
https://github.com/oppia/oppia/blob/c592384dd2327b8fa1aa6d00a4527a32b5609062/core/domain/collection_jobs_one_off_test.py#L52

Other than that, it seems that I need a user to commit the model at:
https://github.com/oppia/oppia/blob/c592384dd2327b8fa1aa6d00a4527a32b5609062/core/domain/collection_jobs_one_off_test.py#L547

Reply:
Hmmm, still worth testing it though? If it won't work I will take a look tomorrow when I'm back home to my main PC.

Reply:
Just tried it and got the test failed again.